lib/runtime-core/src/typed_func.rs

```diff
@@ -246,6 +296,97 @@ where
     }
 }
 
+impl<'a> ErasedFunc<'a> {
+    /// Creates a polymorphic function.
+    #[allow(unused_variables)]
+    #[cfg(all(unix, target_arch = "x86_64"))]
+    pub fn new_polymorphic<F>(signature: Arc<FuncSig>, func: F) -> Self
+    where
+        F: Fn(&mut vm::Ctx, &[crate::types::Value]) -> Vec<crate::types::Value> + 'static,
+    {
+        use crate::trampoline_x64::{CallContext, TrampolineBufferBuilder};
+        use crate::types::Value;
+        use std::convert::TryFrom;
+
```

Comment:
Can we also enforce that `F` is a function pointer or a closure _without_ a captured environment? (meaning: not allowing closure with environments for now)

Something similar to what we are doing in the other `to_raw` functions:

```
                    // `FN` is a function pointer, or a closure
                    // _without_ a captured environment.
                    if mem::size_of::<Self>() == 0 {
                        NonNull::new(&self as *const _ as *mut vm::FuncEnv)
                    }
                    else {
                        unimplemented!("DynamicFunc with captured closures are not implemented");
                    }
```

Reply:
It is not needed for dynamic/polymorphic functions since proper fat pointer `Box<Fn>` calls and drops is implemented: https://github.com/wasmerio/wasmer/blob/31a72e59fbd96263277a5827b3576351b6698ed0/lib/runtime-core/src/typed_func.rs#L313

Reply:
After some internal discussion we decided to temporarily disallow closures with captured environment so that more future changes are possible without breaking the public API.